### ISSUE TITLE
dnsapi for google domains?

### ISSUE THREAD

**Issue by greyltc** [acmesh-official/acme.sh#4107]
Would it be possible to add support google domains dnsapi?

**Comment by Neilpang** [comment]
Is this what you want ?
https://github.com/acmesh-official/acme.sh/wiki/dnsapi#49-use-google-cloud-dns-api-to-automatically-issue-cert

**Comment by greyltc** [comment]
That seems to be some google cloud platform related thing. I'm asking about domains managed via domains.google.com

**Comment by garybuhrmaster** [comment]
> That seems to be some google cloud platform related thing. I'm asking about domains managed via domains.google.com

This has been asked a number of times in other contexts, and the Google product naming adds to the confusion.  Google Domains is a registrar with minimal DNS server functionality, and Google Cloud DNS is a full function DNS solution.  Google Domains does not provide any formal published DNS management API (with the exception of a limited ddns api) although Google Domains does allow you to manage DNS records through a web browser (for some small (website-mostly?) properties that is likely sufficient).  It is possible to use Google Domains as your registrar, and another full featured (API providing) DNS service (including Google Cloud DNS) as your DNS provider.  Support for standalone Google Domains with the acme dnsapi would likely require someone to have to reverse engineer the existing web browser management, and as far as I know no one has does the research/work to determine how to accomplish that (although there are many that seem to have the same itch, no one seems to be scratching).

**Comment by esand** [comment]
Seems Google Domains has API key generation for ACME DNS challenge, and has a link to this project: https://github.com/aaomidi/certbot-dns-google-domains as a certbot plugin. Hopefully this means that it could be added?

**Comment by Neilpang** [comment]
is this what you want ?

https://github.com/acmesh-official/acme.sh/pull/4546

**Comment by NowAnti** [comment]
> is this what you want ?
> 
> #4546

Yes.Google domain now provides API key generation for the ACME domain name challenge.

**Comment by chichilatte** [comment]
Can confirm it works perfectly. 

Just get your `GOOGLEDOMAINS_ACCESS_TOKEN` from Google Domains website (Security > ACME DNS API section). Then follow the simple instructions at https://github.com/acmesh-official/acme.sh/wiki/dnsapi2#157-use-google-domains-dns-api

Fabulous!

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.

**Comment by neilpang** [comment]
Google Domains ACME DNS API support was added in 3.0.6 (2d8c0c01, dns_googledomains.sh); note Google has since discontinued Google Domains.
